### PR TITLE
Adding proxy API's for etherscan

### PR DESCRIPTION
**Pull request by agatsoh**
Hi I am extending the API's for proxy and will help for any other things in my free time. I will also write the remaining API's onceI find time.

### CODE REVIEW

**Comment by agatsoh** [comment]
@corpetty recommended that we stick to python3.6 What is your opinion ?

**Comment by corpetty** [comment]
I agree that sticking with at least >= 3.5 is the way to go.

**Comment by AndreMiras** [comment]
Thanks for asking, my app is running on Python 3.6, but indeed normally libraries should have quite wide support to different python versions. And this is currently enforced with tox checking for 2 and 3 https://github.com/corpetty/py-etherscan-api/blob/f3cd702/tox.ini
But well Python2 should be dropped at some point so I'm fine if we drop it https://pythonclock.org

**Comment by taylorjdawson** [comment]
@corpetty Nope, I am fine if we drop it too.

**Comment by AndreMiras** [comment]
Please also drop Python2.7 from `tox.ini` otherwise Travis will fail https://travis-ci.org/corpetty/py-etherscan-api/jobs/390758838

**Comment by corpetty** [comment]
@agatsoh please remove 2.7 from tox.ini and travis.yaml and re-PR

**Comment by taylorjdawson** [comment]
@agatsoh The naming of 'check_keys_api' is somewhat ambiguous. I feel like adding comments would be valuable here (and eventually required w/ linting integration).

**Comment by taylorjdawson** [comment]
@corpetty Why did that check fail?

File: tests/test_proxies.py [hunk 1]
```diff
@@ -13,8 +13,8 @@ def test_get_most_recent_block(self):
         api = Proxies(api_key=API_KEY)
         # currently raises an exception even though it should not, see:
         # https://github.com/corpetty/py-etherscan-api/issues/32
```
Comment (AndreMiras) [inline tests/test_proxies.py]: We forgot to remove that comment

Comment (AndreMiras) [reply tests/test_proxies.py]: Yes the two lines comments were with a `self.assertRaises(EmptyResponse)` to describe a bug. But the bug was fixed and assert removed in 06f7b0435d2546e3c25938bf3e11f0487b4d6a20, but the comment was left. So yes basically just removing that two comment is enough.
Also you can delete the `print`, I don't think we need `print` in unit tests. Plus I think we should avoid making actual network calls, but actual mock it, but that's a different story.

I'm not sure regarding the rainy scenario. with error code and message.

Comment (AndreMiras) [reply tests/test_proxies.py]: True for the linter it's currently only tests folder, but that one is also failing. e.g. `tests/test_proxies.py:4:1: F401 'etherscan.client.EmptyResponse' imported but unused` https://github.com/corpetty/py-etherscan-api/blob/06f7b04/tests/test_proxies.py#L4 and other couple of more. So really not sure why Travis didn't catch these obvious ones.
They don't pop up when you run `tox` on your dev env?

Comment (taylorjdawson) [reply tests/test_proxies.py]: I just ran flake8?

Comment (AndreMiras) [reply tests/test_proxies.py]: No I ran `tox` that rans `flake8` for you.
```
tox
GLOB sdist-make: /home/andre/workspace/py-etherscan-api/setup.py
pep8 inst-nodeps: /home/andre/workspace/py-etherscan-api/.tox/dist/py_etherscan_api-0.7.0.zip
pep8 installed: certifi==2018.4.16,chardet==3.0.4,flake8==3.5.0,idna==2.6,mccabe==0.6.1,py-etherscan-api==0.7.0,pycodestyle==2.3.1,pyflakes==1.6.0,requests==2.18.4,urllib3==1.22
pep8 runtests: PYTHONHASHSEED='913872812'
pep8 runtests: commands[0] | flake8 tests/
tests/test_proxies.py:4:1: F401 'etherscan.client.EmptyResponse' imported but unused
tests/test_proxies.py:20:1: W391 blank line at end of file
tests/test_proxies.py:20:1: W293 blank line contains whitespace
ERROR: InvocationError: '/home/andre/workspace/py-etherscan-api/.tox/pep8/bin/flake8 tests/'
py3 inst-nodeps: /home/andre/workspace/py-etherscan-api/.tox/dist/py_etherscan_api-0.7.0.zip
py3 installed: atomicwrites==1.1.5,attrs==18.1.0,certifi==2018.4.16,chardet==3.0.4,idna==2.6,more-itertools==4.2.0,pluggy==0.6.0,py==1.5.3,py-etherscan-api==0.7.0,pytest==3.6.2,requests==2.18.4,six==1.11.0,typing==3.6.4,urllib3==1.22
py3 runtests: PYTHONHASHSEED='913872812'
py3 runtests: commands[0] | python -m unittest discover --start-directory=tests/
5848317
/usr/lib/python3.6/unittest/case.py:605: ResourceWarning: unclosed <ssl.SSLSocket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('192.168.1.130', 42938), raddr=('23.111.151.66', 443)>
  testMethod()
./usr/lib/python3.6/unittest/case.py:605: ResourceWarning: unclosed <ssl.SSLSocket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('192.168.1.130', 42940), raddr=('23.111.151.66', 443)>
  testMethod()
./usr/lib/python3.6/unittest/case.py:605: ResourceWarning: unclosed <ssl.SSLSocket fd=3, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('192.168.1.130', 42942), raddr=('23.111.151.66', 443)>
  testMethod()
.
----------------------------------------------------------------------
Ran 3 tests in 2.913s

OK
_____________________________________________________________________________________________________________________________________ summary _____________________________________________________________________________________________________________________________________
ERROR:   pep8: commands failed
  py3: commands succeeded
```

Comment (taylorjdawson) [reply tests/test_proxies.py]: Okay just ran that. Got the same results. When should we switch flake8 to the whole directory?

**Comment by AndreMiras** [comment]
Great! Still don't know why Travis didn't complain about it. It should be
turned on when another pull request fixes the errors :)

**Comment by taylorjdawson** [comment]
We should have a slack/discord. Which would you guys prefer?

**Comment by AndreMiras** [comment]
I was imagining the project to have his own gitter room. From what I remember, all it needs is the maintainer @corpetty to authenticate to https://gitter.im/ and pick a room. Then gitter bot will come with a pull request to add the gitter badge to the project README.
It's better if it's an "official room" created by the maintainer, because a bunch of hooks get added so we see pull requests and ticket updates directly in the room.

**Comment by corpetty** [comment]
done: https://gitter.im/py-etherscan/Lobby#

**Pull request merged**